pkg/data/elasticsearch.go

```diff
@@ -0,0 +1,390 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the AppsCode Community License 1.0.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://github.com/appscode/licenses/raw/1.0.0/AppsCode-Community-1.0.0.md
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package data
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	api "kubedb.dev/apimachinery/apis/kubedb/v1alpha2"
+	cs "kubedb.dev/apimachinery/client/clientset/versioned"
+	"kubedb.dev/apimachinery/pkg/factory"
+	"kubedb.dev/cli/pkg/lib"
+	es_clientgo "kubedb.dev/db-client-go/elasticsearch"
+
+	"github.com/spf13/cobra"
+	shell "gomodules.xyz/go-sh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"kmodules.xyz/client-go/tools/portforward"
+	//"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	idx           = "my-index"
+	alpineCurlImg = "curlimages/curl"
+
+	caFile   = "/tmp/ca.crt"
+	certFile = "/tmp/client.crt"
+	keyFile  = "/tmp/client.key"
+)
+
+type elasticsearchOpts struct {
+	db       *api.Elasticsearch
+	config   *rest.Config
+	client   *kubernetes.Clientset
+	dbClient *cs.Clientset
+	esClient *es_clientgo.Client
+	username string
+	pass     string
+}
+
+func InsertElasticsearchDataCMD(f cmdutil.Factory) *cobra.Command {
+	var (
+		dbName string
+		rows   int
+	)
+
+	esInsertCmd := &cobra.Command{
+		Use: "elasticsearch",
+		Aliases: []string{
+			"es",
+		},
+		Short:   "Insert data to a elasticsearch database",
+		Long:    `Use this cmd to insert data into a elasticsearch database`,
+		Example: `kubectl dba insert -n demo es es-quickstart --rows 1000`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				log.Fatal("Enter elasticsearch object's name as an argument")
+			}
+			dbName = args[0]
+
+			namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
+			if err != nil {
+				klog.Error(err, "failed to get current namespace")
+			}
+
+			config, err := f.ToRESTConfig()
+			if err != nil {
+				log.Fatal("couldn't create config, error: ", err)
+			}
+
+			tunnel, err := lib.TunnelToDBService(config, dbName, namespace, api.ElasticsearchRestPort)
+			if err != nil {
+				log.Fatal("couldn't create tunnel, error: ", err)
+			}
+
+			opts, err := newElasticsearchOpts(config, dbName, namespace, tunnel)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			err = opts.connect(tunnel.Local)
+			if err != nil {
+				log.Fatal(err)
+			}
+			err = opts.insertDataInDatabase(rows)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			tunnel.Close()
```

Comment:
the tunnel always should be closed. Call it with defer just after lib.TunnelToDbService() func